R/transpose.R

```diff
@@ -36,6 +36,16 @@
 #' x <- list(list(a = 1, b = 2), list(a = 3, b = 4), list(a = 5, b = 6))
 #' x %>% transpose()
 #' x %>% transpose() %>% simplify_all()
+#'
+#' # Provide explicit component names to prevent loss of those that don't
+#' # appear in first component
+#' ll <- list(
+#'   list(x = 1, y = "one"),
+#'   list(z = "deux", x = 2)
+#' )
+#' ll %>% transpose()
+#' nms <- ll %>% map(names) %>% reduce(union) %>% unique()
```

Comment:
I don't think you need unique here: the names of each element should already be unique, and `union()` ensures that they stay unique.